### ISSUE TITLE
EnC workspace does not report on generators diagnostics

### ISSUE THREAD

**Issue by jeromelaban** [dotnet/roslyn#70863]
**Version Used**: 4.8.0-3.final

**Steps to Reproduce**:

1. `dotnet new blazorwasm -o testblazor01`
1. `cd testblazor01` 
1. In `testblazor01.csproj` add the following:
  ```xml
  <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
  ```
1.  Run `dotnet watch`
2. In `Program.cs`, add the following at the end:
  ```csharp
   public static partial class Test
   {
       [System.Runtime.InteropServices.JavaScript.JSImport("test")]
       public static partial void Test2(IDisposable test); 
   }
  ```

**Diagnostic Id**: None

**Expected Behavior**: Hot Reload fails with `error SYSLIB1072: The type 'System.IDisposable' is not supported by source-generated JavaScript interop`

**Actual Behavior**: No errors are reported.

**Comment by Youssef1313** [comment]
cc @davidwengier

**Comment by jeromelaban** [comment]
After investigating a bit further, when a solution change is considered by:
https://github.com/dotnet/roslyn/blob/93677976e310e7f910f289e962566485cbabf8f3/src/Features/Core/Portable/EditAndContinue/EditSession.cs#L431

but the compilation fails in generated code, the failing compilation does not get documents or additional documents added to the list returned by `PopulateChangedAndAddedDocumentsAsync`. This causes the workspace update to stop and not report diagnostics here:

https://github.com/dotnet/roslyn/blob/93677976e310e7f910f289e962566485cbabf8f3/src/Features/Core/Portable/EditAndContinue/EditSession.cs#L898-L901

This still has allowed me to find that the delta generation may fail if there are duplicate files in the AdditionalDocuments list.

**Comment by tmat** [comment]
There might be multiple issues here. First is, the diagnostic reported by source generator isn't reported in the IDE:

If you follow the above repro, but instead of dotnet-watch open the project in VS and without running it just add the Test class, the IDE does not show the diagnostics produced by the generator:

**Comment by CyrusNajmabadi** [comment]
Are generator diagnostics something we support?  I'm not sure how that is supposed to flow into pull diagnostics.  Do we need some other IDiagnosticSource for that?

**Comment by tmat** [comment]
@CyrusNajmabadi Turns out we do have `GeneratorDiagnosticsPlaceholderAnalyzer` that is special cased all over `DiagnosticIncrementalAnalyzer` to report the source generator errors :-/

**Comment by tmat** [comment]
VS reports the diagnostic correctly and right away in `Automatic` mode. It `Balanced` mode might take a while.
